core/src/test/java/org/apache/calcite/test/TableInRootSchemaTest.java

```diff
@@ -210,11 +209,27 @@ protected TableWithNullableRowToplevel() {
     Statement statement = calciteConnection.createStatement();
     // Without the fix to this issue the Validator crashes with an AssertionFailure:
     // java.lang.RuntimeException: java.lang.AssertionError:
-    // Conversion to relational algebra failed to preserve datatypes:
-    // validated type:
+    // Conversion to relational algebra failed to preserve datatypes
     ResultSet resultSet = statement.executeQuery("SELECT T.P.K, T.Q.S, T.Q.S.L, T.Q.S.M FROM T");
     resultSet.close();
     statement.close();
     connection.close();
   }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-7043">[CALCITE-7043]
+   * Type inferred for SqlItemOperator has incorrect nullability</a>. */
+  @Test void testNullableRowTopLevel2() throws Exception {
```

Comment:
Can we add test cases in `SqlOperatorTest#testItemOp`?

Reply:
The reason all these tests are here is because the schema of the table needs ROW types that are nullable, and the standard SQL parser does NOT create these correctly, but that is a bug I am not planning to fix (it may be considered a feature). In these tests the schema is created manually using lower level mechanisms. We could do the same in the SqlOperatorTest, but it won't improve the coverage.